web/modules/custom/collection/src/Form/JoinCollectionForm.php

```diff
@@ -130,18 +168,20 @@ public function submitForm(array &$form, FormStateInterface $form_state) {
     /** @var \Drupal\user\UserInterface $user */
     $user = User::load($form_state->getValue('user_id'));
     $role_id = $collection->getEntityTypeId() . '-' . $collection->bundle() . '-' . OgRoleInterface::AUTHENTICATED;
-
+    $state = $collection->get('field_ar_closed')->first()->value ? OgMembershipInterface::STATE_PENDING : OgMembershipInterface::STATE_ACTIVE;
     $membership = OgMembership::create();
     $membership
       ->setUser($user)
       ->setGroup($collection)
-      ->setState(OgMembershipInterface::STATE_ACTIVE)
+      ->setState($state)
       ->setRoles([OgRole::load($role_id)])
       ->save();
 
-    drupal_set_message($this->t('You are now a member of %collection.', [
-      '%collection' => $collection->getName(),
-    ]));
+    $parameters = ['%collection' => $collection->getName()];
+    $message = ($state === OgMembership::STATE_ACTIVE) ?
```

Comment:
Well I removed them but I do not agree.. Parenthesis are meant for grouping. Either visually or logically! Saying that it makes it less readable because it does not change the logic is like saying that it makes it less readable when I split it in multiple lines since I am putting line changes there.